lib/ttdrest/concerns/base.rb

```diff
@@ -20,16 +25,67 @@ def authenticate(options = {})
       def check_response(response)
         if response.code.eql?("403")
```

Comment:
Will this be a string or a symbol?

Should we add other bad response codes, like DNS lookup errors here?

Reply:
I did confirm it is a string.

Do we know what the class of errors for connection errors are? `SocketError`?